airbyte-integrations/connectors/source-mailchimp/source_mailchimp/streams.py

```diff
@@ -138,6 +138,43 @@ def path(self, **kwargs) -> str:
         return "lists"
 
 
+class ListMembers(IncrementalMailChimpStream):
+    cursor_field = "last_changed"
+    data_field = "members"
+
+    def read_records(self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_slice: Mapping[str, Any] = None, stream_state: Mapping[str, Any] = None) -> Iterable[Mapping[str, Any]]:
+        lists_stream = Lists(authenticator=self.authenticator)      
+        for list_record in lists_stream.read_records(sync_mode=SyncMode.full_refresh):
+            list_id = list_record["id"]
+            yield from super().read_records(sync_mode=SyncMode.full_refresh, stream_slice={"list_id": list_id})
+
+    def path(self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None) -> str:
+        list_id = stream_slice["list_id"]
+        return f"lists/{list_id}/members"
+
+
+class GetMemberInfo(IncrementalMailChimpStream):
+    cursor_field = "last_changed"
+
+    def read_records(self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_slice: Mapping[str, Any] = None, stream_state: Mapping[str, Any] = None) -> Iterable[Mapping[str, Any]]:
+        listmembers_stream = ListMembers(authenticator=self.authenticator)        
+        for list_record in listmembers_stream.read_records(sync_mode=SyncMode.full_refresh):
+            subscriber_hash = list_record["email_address"]
+            yield from super().read_records(sync_mode=SyncMode.full_refresh, stream_slice={"email_address": email_address})
+
+    def path(self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None) -> str:
+        subscriber_hash = stream_slice["email_address"]
+        return f"lists/{list_id}/members/{subscriber_hash}"
```

Comment:
Review this implementation of this stream for errors. Use the read command defined in the README.md in the directory of the connector to test that records are being properly fetched.

Reply:
So I have reviewed the stream and made a few changes, the url path now correctly specifies the list_id and subscriber_hash, however it gives me 404 status error. 
my guess is because of the subscriber_hash not being in the MD5 hash format.

Reply:
The reason you're getting a 404 is because list_id is not properly configured for the url. 

My recommendation to you here is to add a print line and print the list_record in the Get Member Info stream. Run the read command and you'll see what I'm talking about, the list_id currently is the MD5 hash of the email address. 

In other words, in this url: `https://us21.api.mailchimp.com/3.0/lists/0bf87600ece53d06366cdedcb4d5c918/members/drakependragon0@gmail.com?`

`0bf87600ece53d06366cdedcb4d5c918` is actually the MD5 hash of `drakependragon0@gmail.com` which forms an incorrect url since it should actually be the list_id

Reply:
You're right! I have checked the docs as well and the id is the MD5 hash of the email address, I should be instead using the "list_id" which will give me the correct url.